### DISCUSSION TITLE
Issue with imageWithLogo.php

### DISCUSSION THREAD

**Discussion by OpusInternet** [chillerlan/php-qrcode#119] [Q&A] [answered]
Hi all and codemasher and congrats for this great library  ! 

I've just installed it localy with composer and I'm running some tests with the examples. So far so good, the QRCodes from image.php and svg.php are displayed just by changing my path to the autoload... 

But I got an issue with imageWithLogo.php. I've copy/pasted the .png to ./images/ and changed its path at the bottom of the php file but when I load the file, the page returns an error : 

`Parse error: syntax error, unexpected 'int' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST) in F:\wamp64\www\tests\QRcodes\imageWithLogo.php on line 26`

The line 26 is "protected int $logoSpaceWidth;" in the class "LogoOptions"

```
class LogoOptions extends QROptions{
	// size in QR modules, multiply with QROptions::$scale for pixel size
	protected int $logoSpaceWidth;
	protected int $logoSpaceHeight;
}
```

Did I do something wrong ? 
Many thanks if anyone can help me on this.

**Comment by codemasher** [comment, marked as answer]
That sounds like you're running v4.x on a PHP lower than 7.4. The error means that your PHP can't interpret the [typed properties](https://www.php.net/manual/en/migration74.new-features.php) which were introduced in 7.4. You can either try v3.4.x which runs from PHP 7.2 onwards or upgrade/switch the PHP version to 7.4 or greater.

**Reply by OpusInternet** [reply]
Many thanks, that's probably why ! I just checked and I use 7.2.10 for now.
I'll make sure I have a 7.4 on the live server.
Thanks again !

**Reply by codemasher** [reply]
Happy to help!
